src/test/java/com/yahoo/sketches/counting/FrequentItemsTest.java

```diff
@@ -0,0 +1,157 @@
+package com.yahoo.sketches.counting;
+
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import java.lang.Math;
+
+/**
+ * Tests FrequentItems class
+ * 
+ * @author edo
+ * 
+ */
+public class FrequentItemsTest {
+
+  @Test
+  public void construct() {
+    int size = 100;
+    FrequentItems frequentItems = new FrequentItems(size);
+    Assert.assertNotNull(frequentItems);
+  }
+
+  @Test
+  public void updateOneTime() {
+    int size = 100;
+    FrequentItems frequentItems = new FrequentItems(size);
+    frequentItems.update(13L);
+    Assert.assertEquals(frequentItems.nnz(), 1);
+  }
+  
+  @Test
+  public void sizeDoesNotGrow() {
+    int maxSize = 100;
+    FrequentItems frequentItems = new FrequentItems(maxSize);
+    for (long key=0L; key<10000L; key++){
+      frequentItems.update(key);
+      Assert.assertTrue(frequentItems.nnz() <= maxSize);
+    }
+  }
+  
+  @Test
+  public void estimatesAreCorectBeofreDeletePhase() {
+    int maxSize = 100;
+    FrequentItems frequentItems = new FrequentItems(maxSize);
+    for (long key=0L; key<99L; key++){
+      frequentItems.update(key);
+      Assert.assertTrue(frequentItems.get(key) == 1);
+      Assert.assertTrue(frequentItems.getMaxError() == 0);
+    }
+  }
+
+  /**
+   * @param prob
+   * @return a number from a geometric distribution
+   */
+  private long randomGeometricDist(double prob){
+    return (long) (Math.log(Math.random()) / Math.log(1.0 - prob));
+  }
+  
+  @Test
+  public void testRandomGeometricDist() {
+    long maxKey = 0L;
+    double prob = .1;
+    for (int i=0; i<100; i++){
+      // Draws a number from a skewed distribution (geometric)
+      long key = randomGeometricDist(prob) ;
+      if (key > maxKey) maxKey = key;
+      //System.out.println(key);
+      Assert.assertTrue((double)maxKey < 20.0/prob);
```

Comment:
Eclipse should be configured to warn that casting from long to double is not necessary (in Java it is considered an automatic promotion).  C++ programmers like them, Java programmers don't.  This is Java. :)  If you want to leave the cast, you should add a @SuppressWarnings("cast") to the method so that it won't show up as a warning for other developers looking at your code and your intention is clear.